cmd/rpcfuzz/argfuzz/argfuzz.go

```diff
@@ -0,0 +1,225 @@
+package argfuzz
+
+import (
+	"encoding/hex"
+	"math/rand"
+	"reflect"
+	"strconv"
+
+	"github.com/google/gofuzz"
+)
+
+func MutateExecutor(arg []byte, c fuzz.Continue) []byte {
+	switch rand.Intn(10) {
+	case 0:
+		return AddRandomCharactersToStart(arg)
+	case 1:
+		return AddRandomCharactersToMiddle(arg)
+	case 2:
+		return AddRandomCharactersToEnd(arg)
+	case 3:
+		return DeleteRandomCharactersInStart(arg)
+	case 4:
+		return DeleteRandomCharactersInMiddle(arg)
+	case 5:
+		return DeleteRandomCharactersInEnd(arg)
+	case 6:
+		return DuplicateBytes(arg)
+	case 7:
+		return AddSpacesRandomly(arg)
+	case 8:
+		return []byte(c.RandString())
+	case 9:
+		return nil
+	default:
+		// Do nothing
+	}
+
+	return arg
+}
+
+func ByteMutator(arg []byte, c fuzz.Continue) []byte {
+	if arg == nil {
+		return arg
+	}
+
+	arg = MutateExecutor(arg, c)
+	// fitty-fitty chance of more mutations
+	if rand.Intn(2) == 0 && arg != nil {
+		return ByteMutator(arg, c)
+	}
+
+	return arg
+}
+
+func MutateRPCArgs(args *[]interface{}, c fuzz.Continue) {
+	for i, d := range *args {
+		if d == nil {
+			continue
+		}
+
+		switch dataType := reflect.TypeOf(d).Kind(); dataType {
+		case reflect.String:
+			(*args)[i] = string(ByteMutator([]byte(d.(string)), c))
+		case reflect.Int:
+			numString := strconv.Itoa(d.(int))
+
+			byteArrString := string(ByteMutator([]byte(numString), c))
+			num, err := strconv.Atoi(byteArrString)
+			if err != nil {
+				(*args)[i] = byteArrString
+			} else {
+				(*args)[i] = num
+			}
+		case reflect.Bool:
+			(*args)[i] = c.RandBool()
+		default:
+			if reflect.TypeOf(d).Kind() == reflect.Ptr {
+				c.Fuzz(d)
+				(*args)[i] = d
+			} else {
+				(*args)[i] = c.RandString()
+			}
+		}
```

Comment:
my line of thinking is to delegate pointers of objects to gofuzz fuzzer (`c.Fuzz`) because it preserves the original types of those variables. for eg.
```
CustomStruct{x: 100, y: "hello"} => gofuzz fuzz => CustomStruct{x: 432413, y: "worldafds"}
```

it's a start, but the end goal here is to implement the fuzzing logic for each of the structs we have, for example with the RPCTestTransactionArgs struct:
```
RPCTestTransactionArgs struct {
    From                 string `json:"from,omitempty"`
    To                   string `json:"to,omitempty"`
    Gas                  string `json:"gas,omitempty"`
    GasPrice             string `json:"gasPrice,omitempty"`
    MaxFeePerGas         string `json:"maxFeePerGas,omitempty"`
    MaxPriorityFeePerGas string `json:"maxPriorityFeePerGas,omitempty"`
    Value                string `json:"value,omitempty"`
    Nonce                string `json:"nonce,omitempty"`
    Data                 string `json:"data"`
}
```

we can write the mutator for it in a way that each field runs through our mutator.